src/components/providers/PushNotificationsProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { Alert, Platform } from "react-native";
+import * as Notifications from "expo-notifications";
+import * as Device from "expo-device";
+
+type PushNotificationsContextState = {
+  isPushEnabled: boolean;
+  setPushEnabled?: (value: boolean) => void;
+  sendPushNotification?: (notificationContent: {
+    title: string;
+    body: string;
+  }) => Promise<void>;
+};
+
+const initialState: PushNotificationsContextState = {
+  isPushEnabled: false,
+};
+
+const PushNotificationsContext = createContext<PushNotificationsContextState>(initialState);
```

Comment:
this will help you not have the function as optional

const PushNotificationsContext = createContext<PushNotificationsContextState | null>(null);